source/extensions/common/aws/credentials_provider_impl.cc

```diff
@@ -195,10 +195,20 @@ std::string getEnvironmentVariableOrDefault(const std::string& variable_name,
 void CredentialsFileCredentialsProvider::refresh() {
   ENVOY_LOG(debug, "Getting AWS credentials from the credentials file");
 
-  const auto credentials_file = getEnvironmentVariableOrDefault(
-      AWS_SHARED_CREDENTIALS_FILE, DEFAULT_AWS_SHARED_CREDENTIALS_FILE);
+  // Default path plus current home directory. Will fall back to / if HOME environment variable does
+  // not exist
+
+  const auto home = getEnvironmentVariableOrDefault("HOME", "");
```

Comment:
FWIW, this seems like a behavior change - An upgrade to this Envoy may change the used credentials file. I'm not sure if it is intended or not, and will let the codeowner @suniltheta decide.
I suggest thinking about moving it to the end, so it won't take precedence over AWS_SHARED_CREDENTIALS_FILE.

Reply:
I believe this change won't make `default_credentials_file_path` take precedence over `AWS_SHARED_CREDENTIALS_FILE`. Env `AWS_SHARED_CREDENTIALS_FILE` is read first on L206 before falling back to using `default_credentials_file_path` as default. Please let me know if I missed something.

For existing users who are upgrading Envoy, the Envoy will look for file in `$HOME/.aws/credentials` instead of `~/.aws/credentials` as a default location, which I think should be ok.

checking if @michaelfinch (original author) has any suggestions.

Reply:
ty for your feedback :) the default location was never working, as `~` is not expanded by Envoy (unless the user had actually created a directory called `~`). This fix was to address that expansion as originally intended but not expected to change behaviour.